api/src/lib/crud.js

```diff
@@ -0,0 +1,128 @@
+import koa from 'koa';
+import coBody from 'co-body';
+import koaRoute from 'koa-route';
+
+export default (queries, restricted, excludedRouteMethod) => {
+    const app = koa();
+    let q;
```

Comment:
`q` meaning isn't obvious. Can we rename `q` to `repositoy` and `queries` to `repositoryFactory` ? I know it's longer but I favor meaningful names.